components/raftstore-v2/src/operation/command/admin/merge/commit.rs

```diff
@@ -323,8 +318,15 @@ impl<EK: KvEngine, R: ApplyResReporter> Apply<EK, R> {
         // TODO: check both are trimmed.
 
         let path = reg.tablet_path(self.region_id(), index);
-        // The last merge is incomplete.
-        if path.exists() {
+        // Check if the last merge is incomplete.
+        // At this point, admin.flushed is smaller than merge index. If `log_recovery`
+        // is None, it means no data cf is flushed after admin.flushed.
+        if path.exists()
+            && self
```

Comment:
What if the condition is false?

Reply:
If it's false, it means at least some data has been flushed in the new tablet, it implies the RocksDB merge operation is completed.